packages/rum-core/src/common/apm-server.js

```diff
@@ -150,7 +155,20 @@ class ApmServer {
         const { status, responseText } = xhr
         reject({ url, status, responseText })
       }
-      xhr.send(payload)
+
+      let canSend = true
+      if (typeof beforeSend === 'function') {
+        canSend = beforeSend(xhr, url, method, headers, payload)
```

Comment:
Is the expectation here that the provided function must be `synchronous`?  If the users provide `async fn` here, our error handling logic would break and result in not capturing those errors.

Reply:
That's correct! see the main comment!